BuildScript.Nuke/Build.Nuget.cs

```diff
@@ -0,0 +1,183 @@
+            GeneratePackagesForNoneSdkProjectWithDebugSymbols(projectFile, config, nugetOutputDirectory);
+        });
+  }
+
+  private void GenerateSinglePackageWithSymbolServerSupport (
+      ProjectMetadata projectFile,
+      string folderSuffix = "")
+  {
+    Configuration.ForEach(
+        config =>
+        {
+          var nugetOutputDirectory = DirectoryHelper.OutputDirectory / c_nugetWithSymbolServerSupportFolderName / (config + folderSuffix);
+          FileSystemTasks.EnsureExistingDirectory(nugetOutputDirectory);
+          if (projectFile.IsSdkProject)
+            GeneratePackagesForSdkProjectWithSymbolServerSupport(projectFile, config, nugetOutputDirectory);
+          else
+            GeneratePackagesForNoneSdkProjectWithSymbolServerSupport(projectFile, config, nugetOutputDirectory);
+        });
+  }
+
+  private void GeneratePackagesForSdkProjectWithDebugSymbols (ProjectMetadata projectFile, string config, AbsolutePath nugetOutputDirectoryPath)
+  {
+    DotNetTasks.DotNetPack(
+        s => s
+            .SetProject(projectFile.Path)
+            .SetConfiguration(config)
+            .SetVersion(VersionHelper.AssemblyNuGetVersion)
+            .SetOutputDirectory(nugetOutputDirectoryPath)
+            .SetIncludeSource(false)
+            .SetIncludeSymbols(false)
+            .SetProperty(c_extraTagsPropertyKey, $"{config}Build ")
+            .SetProperty(c_companyNamePropertyKey, AssemblyMetadata.CompanyName)
+            .SetProperty(c_companyUrlPropertyKey, AssemblyMetadata.CompanyUrl)
+            .SetProperty(c_productNamePropertyKey, AssemblyMetadata.ProductName)
+            .SetProperty(c_assemblyOriginatorKeyFilePropertyKey, DirectoryHelper.SolutionKeyFile)
+            .SetProperty(c_includeReferencedProjectsPropertyKey, "True")
+    );
+  }
+
+  private void GeneratePackagesForSdkProjectWithSymbolServerSupport (
+      ProjectMetadata projectFile,
+      string config,
+      AbsolutePath nugetOutputDirectoryPath)
+  {
+    DotNetTasks.DotNetPack(
+        s => s
+            .SetProject(projectFile.Path)
+            .SetConfiguration(config)
+            .SetVersion(VersionHelper.AssemblyNuGetVersion)
+            .SetOutputDirectory(nugetOutputDirectoryPath)
+            .SetIncludeSource(true)
+            .SetIncludeSymbols(true)
+            .SetSymbolPackageFormat(DotNetSymbolPackageFormat.snupkg)
+            .SetProperty(c_extraTagsPropertyKey, $"{config}Build ")
+            .SetProperty(c_companyNamePropertyKey, AssemblyMetadata.CompanyName)
+            .SetProperty(c_companyUrlPropertyKey, AssemblyMetadata.CompanyUrl)
+            .SetProperty(c_productNamePropertyKey, AssemblyMetadata.ProductName)
+            .SetProperty(c_assemblyOriginatorKeyFilePropertyKey, DirectoryHelper.SolutionKeyFile)
+    );
+  }
+
+  private void GeneratePackagesForNoneSdkProjectWithDebugSymbols (ProjectMetadata projectFile, string config, AbsolutePath nugetOutputDirectoryPath)
+  {
+    NuGetTasks.NuGetPack(
```

Comment:
Doesn't this work for SDK style projects too?

Reply:
Anyway, many properties are the same for `*DebugSymbols` and `*SymbolServer`, these can be setup in a ` NuGetPackSettings SetBaseMetadata(NuGetPackSettings settings)` method, removing the duplication.